pkg/gofr/cmd/terminal/progress_test.go

```diff
@@ -0,0 +1,131 @@
+package terminal
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"gofr.dev/pkg/gofr/testutil"
+)
+
+func TestProgressBar_SuccessCases(t *testing.T) {
+	total := int64(100)
+
+	var out bytes.Buffer
+	stream := &Output{terminal{isTerminal: true, fd: 1}, &out}
+	bar := NewProgressBar(stream, total)
+
+	// Mock terminal size
+	bar.tWidth = 120
+
+	// Increment the progress bar
+	bar.Incr(10)
+
+	// Verify the output
+	expectedOutput := "\r[████░                                             ] 10.000%"
+	if out.String() != expectedOutput {
+		t.Errorf("Unexpected progress bar output: got %q, want %q", out.String(), expectedOutput)
+	}
+
+	// Increment the progress bar to completion
+	bar.Incr(total - 10)
+
+	// Verify the completion output
+	expectedCompletion := "\r[████░                                             ] 10.000%\r" +
+		"[██████████████████████████████████████████████████] 100.000%\n"
+	if out.String() != expectedCompletion {
+		t.Errorf("Unexpected completion output: got %q, want %q", out.String(), expectedCompletion)
+	}
+}
+
+func TestProgressBar_Fail(t *testing.T) {
+	out := testutil.StdoutOutputForFunc(func() {
+		var out bytes.Buffer
+		stream := &Output{terminal{isTerminal: true, fd: 1}, &out}
+		bar := NewProgressBar(stream, int64(-1))
+
+		assert.Equal(t, bar.total, int64(0))
+	})
+
+	assert.Contains(t, out, "error initializing progress bar, total should be > 0")
+}
+
+func TestProgressBar_Incr(t *testing.T) {
+	var out bytes.Buffer
+	stream := &Output{terminal{isTerminal: true, fd: 1}, &out}
+	bar := NewProgressBar(stream, 100)
+	// doing this as while calculating terminal size the code will not
+	// be able to determine it's width since we are not attacting an actual
+	// terminal for testing
+	bar.tWidth = 120
+
+	// Increment the progress by 20
+	b := bar.Incr(int64(20))
+	if !b && bar.current != 20 {
+		t.Errorf("fail: bar incremented value not correct current: %v should be 20", bar.current)
+	}
+
+	expectedOut := "\r[█████████░                                        ] 20.000%"
+	if out.String() != expectedOut {
+		t.Errorf("Unexpected progress bar output: got %q, want %q", out.String(), expectedOut)
+	}
```

Comment:
Here and everywhere else

Why not using ?

require.Equal(t, expectedOut, out.String())